### PR TITLE
Build Mini SBTC time locked (OP_CSV) transactions

### PR DESCRIPTION
**Pull request by radicleart**
Builds a time locked (OP_CSV) transaction. 

With SBTC Mini the users pegin request does not directly transfer BTC to the sbtc wallet. Instead it broadcasts an output that can be spent within a time window (144 Bitcoin blocks) by the SBTC Signers or it can be reclaimed by the sender after that window expires.

The pegin is broadcast and its txid is sent to the Signers binary. The signers have a window of 144 Bitcoin blocks to spend the BTC which in turn triggers the mint of SBTC to the user - their transaction is passed to the .sbtc contract. 

If the signers fail to claim the pegin the user can reclaim the BTC - once the 144 blocks have elapsed.

The initial pegin transaction is outlined below. The pegin data is sent using the OP_DROP format - i.e. the `pegin data` is sent as part of the output that pays the sbtc wallet;

```js
let asmScript = btc.Script.encode([data, 'DROP', 'DUP', 'HASH160', walletUint8, 'EQUALVERIFY', 'CHECKSIG'])
if (isMini) {
	asmScript = btc.Script.encode([
		'IF', 
		     data, 'DROP', 'DUP', 'HASH160', walletUint8, 'EQUALVERIFY', 'CHECKSIG',
		'ELSE', 
		     144, 'CHECKSEQUENCEVERIFY', 'DROP', btc.OutScript.encode(btc.Address(this.net).decode(this.fromBtcAddress)),
		    'CHECKSIG',
		'ENDIF'
	])
}
```

### CODE REVIEW

**Comment by radicleart** [comment]
An example of this transaction can be found on mempool;

**Comment by sergey-shandar** [comment]
@igorsyl it would be good to have a link on a preview, for example, like CloudFlare preview.

**Comment by sergey-shandar** [comment]
> @igorsyl it would be good to have a link on a preview, for example, like CloudFlare preview.

Oh, I've found it in Checks/Cloudflare Pages tab https://81-support-sbtc-mini-time-lo.sbtc-bridge.pages.dev/

File: src/lib/domain/PegInTransaction.ts [hunk 1]
```diff
@@ -149,40 +146,114 @@ export default class PegInTransaction extends PegTransaction implements PegInTra
...
+				btc.OutScript.encode(btc.Address(this.net).decode(this.pegInData.sbtcWalletAddress)),
+				'ELSE', 
+				144, 'CHECKSEQUENCEVERIFY', 'DROP',
+				btc.OutScript.encode(btc.Address(this.net).decode(this.fromBtcAddress)),
+				'CHECKSIG',
+				'ENDIF'
+			])
+			tx.addOutput({ script: asmScript, amount: BigInt(this.pegInData.amount) });
+		} else {
+			tx.addOutputAddress(this.pegInData.sbtcWalletAddress, BigInt(this.pegInData.amount), this.net);
```
Comment (netrome) [inline src/lib/domain/PegInTransaction.ts:182]: Wait, this function is named `buildOpReturnTransaction`, I guess it should use the old format and not the timelocked OP_DROP variant?

**Review by netrome — APPROVED** [review]
One question I'd love to sync with you, but otherwise things generally look good to me. You're spearheading the commit-reveal ops. I have some catching up to do 😅

**Comment by radicleart** [comment]
Closing without merging this PR. It is out of date.

**Pull request closed without merging**